app/scripts/controllers/network/network-controller.js

```diff
@@ -239,23 +250,25 @@ export default class NetworkController extends EventEmitter {
     return rpcUrl;
   }
 
-  setRpcTarget(rpcUrl, chainId, ticker = 'ETH', nickname = '', rpcPrefs) {
-    assert.ok(
-      isPrefixedFormattedHexString(chainId),
-      `Invalid chain ID "${chainId}": invalid hex string.`,
-    );
-    assert.ok(
-      isSafeChainId(parseInt(chainId, 16)),
-      `Invalid chain ID "${chainId}": numerical value greater than max safe value.`,
-    );
+  /**
+   * A method for setting a network provider by UUID.
+   *
+   * @param {string} uuid - the universal unique identifier that corresponds to the network configuration we wish to set as selected.
+   * @returns {Promise<string>} The RPC Target URL confirmed.
+   */
+  setNetworkTarget(uuid) {
+    if (typeof uuid === 'object') {
+      uuid = uuid.uuid;
+    }
```

Comment:
this is an unfortunate byproduct of the pattern [here](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js#L269) where we pass the request data to both the request template and the method. Probably can make this better.

Reply:
@Gudahtt do you have any thoughts on how this might be improved?

Reply:
nvm. Resolved [here](https://github.com/MetaMask/metamask-extension/pull/17421/commits/930d47f53d5c35b5b6ef1ffa1c711dbce1bef125#diff-b3485a1ea9943fd380364f73a53295289f2ccd3817423d4f245b5102f8620b3cR161-R173)